scipy/stats/_resampling.py

```diff
@@ -2235,30 +2235,34 @@ class PermutationMethod(ResamplingMethod):
     def random_state(self):
         # Uncomment in SciPy 1.17.0
         # warnings.warn(_rs_deprecation, DeprecationWarning, stacklevel=2)
-        return self._rng
+        return self._random_state
 
     @random_state.setter
     def random_state(self, val):
         # Uncomment in SciPy 1.17.0
         # warnings.warn(_rs_deprecation, DeprecationWarning, stacklevel=2)
-        self._rng = val
+        self._random_state = val
 
     @property  # type: ignore[no-redef]
     def rng(self):  # noqa: F811
         return self._rng
 
-    @random_state.setter
-    def rng(self, val):  # noqa: F811
-        self._rng = np.random.default_rng(val)
-
-    @_transition_to_rng('random_state', position_num=3, replace_doc=False)
-    def __init__(self, n_resamples=9999, batch=None, rng=None):
-        self._rng = rng  # don't validate with `default_rng` during SPEC 7 transition
+    def __init__(self, n_resamples=9999, batch=None, random_state=None, *, rng=None):
+        # Uncomment in SciPy 1.17.0
+        # warnings.warn(_rs_deprecation.replace('attribute', 'argument'),
+        #               DeprecationWarning, stacklevel=2)
+        self._rng = rng
+        self._random_state = random_state
```

Comment:
For `_random_state` do we also need to add an equivalent to `_rng: object = field(init=False, repr=False, default=None)  # type: ignore[assignment]` above?

Reply:
With all the customization,I think  the only thing that part does is make it part of the `repr`. I left it out because I think that it's optional and undesirable if we want people to move away from using it.

Reply:
Ah yes confirmed locally this is fine